src/test/rpc/NoRippleCheck_test.cpp

```diff
@@ -258,22 +258,37 @@ class NoRippleCheckLimits_test : public beast::unit_test::suite
         {
             if (! admin)
             {
-                // endpoint drop prevention. Non admin ports will drop requests
-                // if they are coming too fast, so we manipulate the resource
-                // manager here to reset the enpoint balance (for localhost) if
-                // we get too close to the drop limit.
+                // this is endpoint drop prevention. Non admin ports will drop
+                // requests if they are coming too fast, so we manipulate the
+                // resource manager here to reset the enpoint balance (for
+                // localhost) if we get too close to the drop limit. It would
+                // be better if we could add this functionality to Env somehow
+                // or otherwise disable endpoint charging for certain test
+                // cases.
+
+                // keep track of the last endoint balance so that we know
+                // when we are approaching the dropThreshold limit.
+                static int lastBalance;
```

Comment:
The `if (i == 0)`condition suggests that `lastBalance` doesn't really need to be a `static`.  Consider lofting `lastBalance` above the `for` and making it a local.